lobe_android/app/src/main/java/com/example/test/tflite/Classifier.java

```diff
@@ -133,12 +133,21 @@ protected Classifier(Activity activity, Device device, int numThreads) throws IO
         tfliteOptions.setNumThreads(numThreads);
         tflite = new Interpreter(tfliteModel, tfliteOptions);
 
-        // Loads labels out from the signature file.
-        String json = new String(FileUtil.loadByteFromFile(activity, getLabelPath()), StandardCharsets.UTF_8);
-        JSONObject jsonObject = null;
+
+        // Defaults
         labels = new ArrayList<String>();
+        int tempImageSizeY = 0;
+        int tempImageSizeX = 0;
+        DataType imageDataType = DataType.FLOAT32;
+        DataType probabilityDataType = DataType.FLOAT32;
+        int[] probabilityShape = new int[0];
+
+        // Load signature file to obtain labels, input, and output formats of the model.
+        // Using this instead of tflite.get* methods as order of operations and fields are not guaranteed.
         try {
-            jsonObject = new JSONObject(json);
+            // Loads labels out from the signature file.
+            String json = new String(FileUtil.loadByteFromFile(activity, getLabelPath()), StandardCharsets.UTF_8);
+            JSONObject jsonObject = new JSONObject(json);
```

Comment:
might be easier to use the Jackson library for JSON parsing -- I have that in our Java local API snippet

Reply:
Yeah, I'm sure it could be better. I'm also okay doing that in a follow up since this is tested and working for now :D